packages/expo-in-app-purchases/android/src/main/java/expo/modules/inapppurchases/InAppPurchasesModule.java

```diff
@@ -0,0 +1,134 @@
+package expo.modules.inapppurchases;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+
+import android.content.Context;
+import android.app.Activity;
+
+import com.android.billingclient.api.BillingClient;
+import com.android.billingclient.api.Purchase;
+
+import org.unimodules.core.ExportedModule;
+import org.unimodules.core.ModuleRegistry;
+import org.unimodules.core.Promise;
+import org.unimodules.core.interfaces.ExpoMethod;
+import org.unimodules.core.interfaces.ActivityProvider;
+import org.unimodules.core.interfaces.ModuleRegistryConsumer;
+
+public class InAppPurchasesModule extends ExportedModule implements ModuleRegistryConsumer {
+  private static final String TAG = InAppPurchasesModule.class.getSimpleName();
+  private static final String NAME = "ExpoInAppPurchases";
+
+  private BillingManager mBillingManager;
+  private ModuleRegistry mModuleRegistry;
+
+  public InAppPurchasesModule(Context context) {
+    super(context);
+  }
+
+  @Override
+  public String getName() {
+    return NAME;
+  }
+
+  @Override
+  public void setModuleRegistry(ModuleRegistry moduleRegistry) {
+    mModuleRegistry = moduleRegistry;
+  }
+
+  @Override
+  public Map<String, Object> getConstants() {
+    Map<String, Object> constants = new HashMap<>();
+
+    constants.put("responseCodes", getBillingResponseCodes());
+    constants.put("purchaseStates", getPurchaseStates());
+
+    return constants;
+  }
+
+  @ExpoMethod
+  public void connectToAppStoreAsync(final Promise promise) {
+    Activity activity = getCurrentActivity();
+    mBillingManager = new BillingManager(activity);
```

Comment:
Imho we should check if activity is not `null` here and maybe reject otherwise? the current activity may not be present if you run this code when the app is backgrounded

Reply:
I agree. I wasn't sure which use cases activity might be null but that makes sense.